liblwgeom/lwgeom_geos.c

```diff
@@ -1233,6 +1244,83 @@ lwgeom_offsetcurve(const LWLINE* lwline, double size, int quadsegs, int joinStyl
 	return result;
 }
 
+static LWGEOM*
+lwgeom_offsetcurve_multilinestring(const LWGEOM* geom, double size, int quadsegs, int joinStyle, double mitreLimit)
+{
+	int32_t srid = get_result_srid(geom, NULL, __func__);
+	uint8_t is3d = FLAGS_GET_Z(geom->flags);
+	if (srid == SRID_INVALID) return NULL;
+
+	if (!lwgeom_is_collection(geom))
+		return lwgeom_offsetcurve_line(lwgeom_as_lwline(geom), size, quadsegs, joinStyle, mitreLimit);
+	else
+	{
+		LWCOLLECTION* result;
+		LWGEOM* tmp;
+		const LWCOLLECTION* col = lwgeom_as_lwcollection(geom);
+		int i;
+		result = lwcollection_construct_empty(MULTILINETYPE, srid, is3d, LW_FALSE);
+		for (i = 0; i < col->ngeoms; i++)
+		{
+			tmp = lwgeom_offsetcurve_line(
+			    lwgeom_as_lwline(col->geoms[i]), size, quadsegs, joinStyle, mitreLimit);
+
+			if (!tmp)
+			{
+				lwcollection_free(result);
+				return NULL;
+			}
+
+			if (lwgeom_get_type(tmp) != LINETYPE && lwgeom_get_type(tmp) != MULTILINETYPE) return NULL;
+
+			if (!lwgeom_is_empty(tmp))
+			{
+				if (!lwgeom_is_collection(tmp))
+					lwcollection_add_lwgeom(result, tmp);
+				else
+				{
+					const LWCOLLECTION* c2 = lwgeom_as_lwcollection(tmp);
```

Comment:
Consider a comment to explain what's happening in this block

Reply:
made separate function for collection inline concatenation